src/object/cli_shard.c

```diff
@@ -357,25 +357,26 @@ obj_ec_iom_merge(struct obj_reasb_req *reasb_req, uint32_t shard,
 	daos_recx_t		 hi, lo, recx, tmpr;
 	daos_recx_t		 recov_hi, recov_lo;
 	uint32_t		 iom_nr, i;
-	bool			 done;
+	bool			 done, is_data_shard;
 	int			 rc = 0;
 
 	D_ASSERT(tgt_idx < obj_ec_data_tgt_nr(oca));
 
 	if (recov_list != NULL)
 		daos_recx_ep_list_hilo(recov_list, &recov_hi, &recov_lo);
 
+	is_data_shard = (shard % obj_ec_tgt_nr(oca)) < obj_ec_data_tgt_nr(oca);
 	D_SPIN_LOCK(&reasb_req->orr_spin);
 
 	/* merge iom_recx_hi */
 	hi = src->iom_recx_hi;
 	end = DAOS_RECX_END(hi);
-	if (end > 0)
+	if (end > 0 && is_data_shard) {
```

Comment:
I thought for parity shard, end is always 0 here

Reply:
when hit some exts (or all fetch recxs hit) on replica space, the hi, lo, and iom_recxs are all valid, no?

Reply:
Hmm, I am confused here.  so this is trying to get the highest data cells?  then why skip it for the extent from parity node?

Reply:
if it is from parity shard, then it is repilca ext that is in daos offset already right? so need not the converting.

Reply:
I thought this is also choosing the max from all shards. (data or parity shard)

 hi.rx_idx = max(hi.rx_idx, rounddown(end - 1, cell_rec_nr));

No?